pkg/credentials/jwt_source_provider.go

```diff
@@ -1,17 +1,27 @@
 package credentials
 
 import (
+	"github.com/spiffe/go-spiffe/v2/logger"
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
 )
 
-type JWTSourceProvider func(subject spiffeid.ID, workloadSocket string, audience string) JWTSource
+type JWTSourceProvider func(
+	subject spiffeid.ID,
+	workloadSocket string,
+	audience string,
+	logger logger.Logger) JWTSource
 
-func StandardJWTSourceProvider(subject spiffeid.ID, workloadSocket string, audience string) JWTSource {
-	return NewJWTSVIDSource(subject, workloadSocket, audience)
+func StandardJWTSourceProvider(
+	subject spiffeid.ID,
+	workloadSocket string,
+	audience string,
+	logger logger.Logger) JWTSource {
+
+	return NewJWTSVIDSource(subject, workloadSocket, audience, logger)
 }
 
-func StaticJWTSourceProvider(source JWTSource) func(spiffeid.ID, string, string) JWTSource {
-	return func(spiffeid.ID, string, string) JWTSource {
+func StaticJWTSourceProvider(source JWTSource) func(spiffeid.ID, string, string, logger.Logger) JWTSource {
+	return func(spiffeid.ID, string, string, logger.Logger) JWTSource {
 		return source
 	}
 }
```

Comment:
Adding a Logger as a parameter. Note that this is the SPIFFE Logger and not the Logrus Logger, although Logrus is compatible with the SPIFFE Logger's interface. If we want to push the SPIFFE logger down farther and not expose it via the signature, I wouldn't be against it.